src/services/test_results/TestResultsBuilder.ts

```diff
@@ -1,64 +1,154 @@
+    }
+
+    public static buildGherkinTestResult(testResults: GherkinResultData[], server_id: string, job_id: string, build_id: string, logger: LogUtils): TestResult {
+        if (!testResults || !testResults.length) {
+            logger.info('No gherkin test results were retrieved/found');
+            return null;
+        }
+
+        let testResultTestRuns = this.buildGherkinResultElement(testResults);
+        let testResultTestFields = this.buildTestResultTestFields(this.CUCUMBER_TEST_TYPE);
+        let testResultBuild = new TestResultBuildAttributes(server_id, job_id, build_id);
+
         return new TestResult(testResultBuild, testResultTestFields, testResultTestRuns);
     }
 
-    public static getTestResultXml(testResults: any, server_id: string, job_id: string, logger: LogUtils): any {
-        let result: TestResult = TestResultsBuilder.buildTestResult(testResults, server_id, job_id, logger);
+    private static getTestResultXml(testResult: TestResult, logger: LogUtils): string {
         let options = {compact: true, ignoreComment: true, spaces: 4};
-        let convertedXml = '<?xml version="1.0" encoding="utf-8" standalone="yes"?><test_result>' + convert.js2xml(result, options) + '</test_result>';
+        let convertedXml = '<?xml version="1.0" encoding="utf-8" standalone="yes"?><test_result>' + convert.js2xml(testResult, options) + '</test_result>';
         logger.info('Test results converted for Octane');
         logger.debug(convertedXml);
         return convertedXml;
     }
 
-    public static async getTestsResultsByBuildId(connection: WebApi, projectName: string, buildId: number, serverId: string, jobId: string, logger: LogUtils): Promise<string> {
+    private static getUnitTestResultXml(testResults: TestCaseResult[], server_id: string, job_id: string, logger: LogUtils): string {
+        let result: TestResult = TestResultsBuilder.buildUnitTestResult(testResults, server_id, job_id, logger);
+        return this.getTestResultXml(result, logger);
+    }
+
+    private static getGherkinTestResultXml(testResults: GherkinResultData[], server_id: string, job_id: string, build_id: string, logger: LogUtils): string {
+        let result: TestResult = TestResultsBuilder.buildGherkinTestResult(testResults, server_id, job_id, build_id, logger);
+        return this.getTestResultXml(result, logger);
+    }
+
+    public static async getTestsResultsByBuildId(connection: WebApi, projectName: string, buildId: number, serverId: string, jobId: string, cucumberReportsPath: string, logger: LogUtils): Promise<string[]> {
+        let xmlTestResults: string[] = [];
+        let gherkinResults = this.getCucumberReportsFromPath(cucumberReportsPath, logger);
         let buildApi: ba.IBuildApi = await connection.getBuildApi();
         let build = await buildApi.getBuild(projectName, buildId);
         let buildURI = build.uri;
         let testApi: ta.ITestApi = await connection.getTestApi();
+
+        if (gherkinResults.length > 0) {
```

Comment:
gherkinResults are processed in a single line, while testRuns have a whole block in here. This is confusing. Can this be homogenised somehow, to look similar and be easier to understand?